packages/toolpad-core/src/jsServerRuntime.tsx

```diff
@@ -27,10 +27,9 @@ export function createServerJsRuntime(env?: Record<string, string | undefined>):
   };
 }
 
-export function useServerJsRuntime(): JsRuntime {
+export function useServerJsRuntime(processEnv: Record<string, string | undefined> = {}): JsRuntime {
   return React.useMemo(() => {
     // process.env is not available in the browser
-    const processEnv = {};
     return createServerJsRuntime(processEnv);
-  }, []);
+  }, [processEnv]);
```

Comment:
I don't think we can do this...
The runtime cannot have access to the environment variables, they should only be accessible on the server.